x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/user_flyout.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiIcon,
+  EuiText,
+  EuiTitle,
+  EuiSpacer,
+} from '@elastic/eui';
+
+interface Props {
+  children: React.ReactNode;
+  isNew: boolean;
+  isComplete: boolean;
+  disabled: boolean;
+  closeUserFlyout(): void;
+  handleSaveUser(): void;
+}
+
+import { CANCEL_BUTTON_LABEL, CLOSE_BUTTON_LABEL } from '../constants';
+
+import {
+  USERS_HEADING_LABEL,
+  UPDATE_USER_LABEL,
+  ADD_USER_LABEL,
+  USER_ADDED_LABEL,
+  USER_UPDATED_LABEL,
+  NEW_USER_DESCRIPTION,
+  UPDATE_USER_DESCRIPTION,
+} from './constants';
+
+export const UserFlyout: React.FC<Props> = ({
```

Comment:
This is tough to QA without the Users Table+Flyout being added to the Users & Roles view - can we include that in this PR as well?

Aslo, while I'm here - do we want to include renaming `Users & roles` -> `Users and roles` as part of this PR, if we're rolling everything into one?

Reply:
Sorry, I missed the PR description of this being PR 1/3 - I think I got confused by an earlier Slack comment about rolling everything in 1 PR. Going to just collapse this comment